core-it-support/maven-it-plugin-bootstrap/src/main/java/org/apache/maven/its/bootstrap/DownloadMojo.java

```diff
@@ -104,6 +105,21 @@ public void execute() throws MojoFailureException
             }
         }
 
+        // HACK: this plugin is executing in bootstrap project that has packaging=pom, but still enlists install/deploy
+        try
+        {
+            for ( Plugin plugin : session.getCurrentProject().getBuildPlugins() )
+            {
+                dependencies.add( toDependency(
```

Comment:
What about  make `dependencies`  as a set to avoid duplicates?

Reply:
Resolving when once done is fast: basically presence of file(s) on disk happens, so duplication (for now) is not something I'd worry about